### ISSUE TITLE
Should putting `@observable` on a getter produce an error?

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [mobxjs/mobx#1454]
I have a:

1. [x] Question: Feel free to just state your question. For a quick answer, there are usually people online at our [Gitter](https://gitter.im/mobxjs/mobx) channel.

I did a silly:

```typescript
class MyStore {
    @observable
    public value: number;

    @observable // should have been @computed
    public get double() { return this.value * 2; }
}
```

Should this produce an error? In practice, `MyStore.prototype.timeSinceLastUpdated` is `undefined`, as is all its instances.

**Comment by mweststrate** [comment]
It is not a supported case indeed, so a PR for runtime checks is welcome

**Comment by quanganhtran** [comment]
Released in 4.2.1
